src/core/qgsmaplayer.h

```diff
@@ -133,6 +135,52 @@ class CORE_EXPORT QgsMapLayer : public QObject
     Q_DECLARE_FLAGS( LayerFlags, LayerFlag )
     Q_FLAG( LayerFlags )
 
+    /**
+     * Categories of style to distinguish appropriate sections for import/export
+     * \since QGIS 3.4
+     */
+    enum StyleCategory
+    {
+      LayerConfiguration = 1 << 0, //!< Flags, display expression, read-only
+      Symbology          = 1 << 1,
```

Comment:
Can you add dox for all these? I realise some are self explanatory, but it's the precedent